src/table/use-column-widths.tsx

```diff
@@ -145,8 +139,7 @@ export function ColumnWidthsProvider({ visibleColumns, resizableColumns, contain
     updateColumnWidths();
   });
 
-  // The widths of the dynamically added columns (after the first render) if not set explicitly
-  // will default to the DEFAULT_COLUMN_WIDTH.
+  // Re-calculate widths of any "new" columns - either on first render, or dynamically added
   useEffect(() => {
```

Comment:
It would be great if we can add more code comments explaining what is actually happening inside as it is quite hard to read. For instance, there are calls like updateColumnWidths(), setColumnWidth(...), readWidth(...) - and what exactly happening is not obvious.